libnmstate/state.py

```diff
@@ -258,11 +277,22 @@ def merge_route_config(self, other_state):
                     self_iface_route_sets[route[Route.NEXT_HOP_INTERFACE]].add(
                         RouteEntry(route))
 
+        # Check whether user desire routes to down interface/ip.
+        _validate_routes(self_iface_route_sets,
+                         iface_enable_states,
+                         ipv4_enable_states,
+                         ipv6_enable_states)
+
         for route in other_state._config_routes:
             # The other_state(running state) does not have any absent
             # route entry and all have NEXT_HOP_INTERFACE
-            iface_name = route[Route.NEXT_HOP_INTERFACE]
-            self_iface_route_sets[iface_name].add(RouteEntry(route))
+            if _route_is_valid(route,
+                               iface_enable_states,
+                               ipv4_enable_states,
+                               ipv6_enable_states):
+                # Don't copy route from down interface or disabled IP
```

Comment:
I do not understand the comment. Is it meant for the if statement? IMHO it can be removed since `_route_is_valid()` describes what is dows in its doctext IIRC. At least it should instead of having the comment here if it needs explanation.

Reply:
Oh. I should add more comments there.

The `_route_is_valid()` is make sure nmstate is not copying invalid route from current state. It just ignore invalid routes.

The `_validate_routes()` is validating __user__ desire state which will raise exception for invalid route.

Reply:
my comment is outdated since the commit was merged in #363